src/elementary/src/join.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef NJOY_ELEMENTARY_JOIN
+#define NJOY_ELEMENTARY_JOIN
+
+// system includes
+#include <numeric>
+#include <string>
+
+// other includes
+
+namespace njoy {
+namespace elementary {
+
+  /**
+   *  @brief Join a list of string to a single string using a given separator
+   *
+   *  @param[in] strings     the strings to be joined together
+   *  @param[in] separator   the separator to be used
+   */
+  std::string join( const std::vector< std::string >& strings,
```

Comment:
I like this `join` function. We should (at some point) move this into [utility](https://github.com/njoy/utility).